spec/controllers/projects_controller_spec.rb

```diff
@@ -1,13 +1,85 @@
 require 'spec_helper'
 
+RSpec.shared_examples 'denies access' do
+  before { request }
+
+  it 'redirects to the root path' do
+    expect(response).to redirect_to(root_path)
```

Comment:
I don't think this is something we should be testing, It's behaviour that cancancan gives us, not something we have explicitly added. The general rule of thumb is to only test the behaviour of the code we have written and not what third part code does. This would be more suited to a integration test.